lib/std/math/modf.zig

```diff
@@ -1,207 +1,143 @@
-// Ported from musl, which is licensed under the MIT license:
-// https://git.musl-libc.org/cgit/musl/tree/COPYRIGHT
-//
-// https://git.musl-libc.org/cgit/musl/tree/src/math/modff.c
-// https://git.musl-libc.org/cgit/musl/tree/src/math/modf.c
-
 const std = @import("../std.zig");
 const math = std.math;
 const expect = std.testing.expect;
 const expectEqual = std.testing.expectEqual;
-const maxInt = std.math.maxInt;
+const expectApproxEqAbs = std.testing.expectApproxEqAbs;
 
-fn modf_result(comptime T: type) type {
+pub fn Modf(comptime T: type) type {
     return struct {
         fpart: T,
         ipart: T,
     };
 }
-pub const modf32_result = modf_result(f32);
-pub const modf64_result = modf_result(f64);
 
 /// Returns the integer and fractional floating-point numbers that sum to x. The sign of each
 /// result is the same as the sign of x.
+/// In comptime, may be used with comptime_float
 ///
 /// Special Cases:
 ///  - modf(+-inf) = +-inf, nan
 ///  - modf(nan)   = nan, nan
-pub fn modf(x: anytype) modf_result(@TypeOf(x)) {
-    const T = @TypeOf(x);
-    return switch (T) {
-        f32 => modf32(x),
-        f64 => modf64(x),
-        else => @compileError("modf not implemented for " ++ @typeName(T)),
+pub fn modf(x: anytype) Modf(@TypeOf(x)) {
+    @setFloatMode(.strict);
+    var result = Modf(@TypeOf(x)){
+        .ipart = @trunc(x),
+        .fpart = undefined,
```

Comment:
You can avoid mutable variables here:
```suggestion
    const ipart = @trunc(x);
    return .{
        .ipart = ipart,
        .fpart = x - ipart,
    };
```